be/src/exec/pipeline/scan/chunk_source.cpp

```diff
@@ -41,77 +42,54 @@ void ChunkSource::unpin_chunk_token() {
     _chunk_token.reset(nullptr);
 }
 
-Status ChunkSource::buffer_next_batch_chunks_blocking(size_t batch_size, RuntimeState* state) {
-    if (!_status.ok()) {
-        return _status;
-    }
+std::pair<Status, size_t> ChunkSource::buffer_next_batch_chunks_blocking(RuntimeState* state, size_t batch_size,
```

Comment:
can we use StatusOr

Reply:
Maybe not, because when it returns non-ok status, we also need the value. However, `StatusOr` cannot reach this.